### DISCUSSION TITLE
Expose as platformio module

### DISCUSSION THREAD

**Discussion by mihaiblaga89** [ESPresense/ESPresense#8] [Ideas]
I think it would be a nice addition to expose this as a platformio module, so people running ESPHome can just import it there directly. I, for example, have around 10 ESPHome devices around the house and I would like to use those as trackers, not deploy new ones.

**Comment by DTTerastar** [comment]
I totally agree and would like it as well.  I've looked into it a bit and I still am trying to understand how esphome components wire themselves in to even tell if it's possible.

**Reply by mihaiblaga89** [reply]
Had the same questions while trying to implement support for a custom sensor that ESPHome didn't have. Apparently all you need to do is either init the required lib in your code or declare it in esphome and use it directly. For example I was using i2c in my lib and I needed to define it in my esphome config so it would load and init the library so I was able to use it. They also expose a "priority" override so you can define when to init your code so you have everything available beforehand. While I don't have that much experience with C++ maybe I can help in one way or another

**Comment by stephenmetzger** [comment]
This would be great. I currently have a number of ESP8266 motion sensors around my house and would love to swap out the 8266's for ESP32s and run both motion detection and presence.

**Reply by DTTerastar** [reply]
I'm moving all the btle stuff into a separate lib... Making some progress towards this possibility...

**Reply by stephenmetzger** [reply]
Thanks a ton!
